src/parser/input-impl.h

```diff
@@ -100,7 +100,7 @@ inline std::optional<uint64_t> ParseInput::takeOffset() {
       if (subLexer == subLexer.end()) {
         return {};
       }
-      if (auto o = subLexer->getU64()) {
+      if (auto o = subLexer->getU<uint64_t>()) {
```

Comment:
`getU<uint64_t>` mentions "unsigned" twice. Can this be `get<uint64_t>`? (maybe using https://en.cppreference.com/w/cpp/types/is_unsigned)

Reply:
`getU` and `getI` both return unsigned integers, so I think it's good to differentiate them in the method name and not favor one over the other (and all the other `getXXX` methods) by giving it the shorter `get` name.

One alternative I considered was to make the template parameter just `32` or `64`, etc. but that would require more complex machinery in the implementation and the nicer API didn't seem worth the extra complexity. If you're interested, I can add a commit with that design so we can see the difference, though.

Reply:
Ah, maybe I've forgotten what those functions mean. I don't see comments on

https://github.com/WebAssembly/binaryen/blob/845e0700c99d603c920d82c2312ec9b39a555c68/src/parser/lexer.h#L128-L135

Perhaps it's worth adding some? Reading the code, the meaning seems to be "U is unsigned, S is signed, I accepts both as inputs but returns unsigned" - ?

Reply:
Yes, that's right. `uN`, `sN`, and `iN` come directly from the grammar in the spec, but I can add comments here as well.

Reply:
Right, sorry, I keep reading this code as normal C++ and I forget it mirrors the spec...